crates/matrix-sdk/src/event_cache/mod.rs

```diff
@@ -312,6 +316,7 @@ impl EventCache {
 }
 
 type AllEventsMap = Arc<RwLock<BTreeMap<OwnedEventId, (OwnedRoomId, SyncTimelineEvent)>>>;
+type RelationsMap = Arc<RwLock<BTreeMap<OwnedEventId, BTreeSet<OwnedEventId>>>>;
```

Comment:
There are two locks for data that's touched around the same time; I'd suggest grouping the two fields inside a data structure, and using a single lock, to avoid having to think about lock ordering.